wasmbinding/stargate_query_test.go

```diff
@@ -0,0 +1,143 @@
+package wasmbinding_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+
+	"github.com/NibiruChain/nibiru/wasmbinding"
+
+	"github.com/NibiruChain/nibiru/x/common/set"
+
+	devgas "github.com/NibiruChain/nibiru/x/devgas/v1/types"
+	epochs "github.com/NibiruChain/nibiru/x/epochs/types"
+	inflation "github.com/NibiruChain/nibiru/x/inflation/types"
+	oracle "github.com/NibiruChain/nibiru/x/oracle/types"
+	sudotypes "github.com/NibiruChain/nibiru/x/sudo/types"
+	tokenfactory "github.com/NibiruChain/nibiru/x/tokenfactory/types"
+)
+
+/*
+TestWasmAcceptedStargateQueries: Verifies that the query paths registered in
+the Wasm keeper's StargateQuerier are the official method names in the gRPC
+query service of each path's respective module.
+
+> ℹ️  "All stargate query paths must be actual GRPC query service methods"
+
+Please see the function doc comment for WasmAcceptedStargateQueries in
+stargate_query.go to understand in detail what invariants this test checks
+for.
+
+Given only the `PB_MSG.PACKAGE` and the `PB_MSG.NAME` of either the query
+request or response, we should know the `QueryRequest::Stargate.path`
+deterministically.
+*/
+func TestWasmAcceptedStargateQueries(t *testing.T) {
+	stargateQueryPaths := set.New[string]()
+	t.Log("stargateQueryPaths: Add cosmos and ibc query paths")
+	// These are hard-coded because the GRPC service descriptions aren't exported
+	// from the Cosmos-SDK and remain private vars. Maybe we could ask the
+	// maintainers to export them in the future.
+	for _, queryPath := range []string{
```

Comment:
Maybe it's better to iterate over the return map of `WasmAcceptedStargateQueries()` instead of having a duplicate list of hard-coded strings.

Reply:
Good idea. It makes it more concise too. Updated in 6f4b57767b63a222982e1e2bfa27b215e8d7c570